File: usenix80/toronto/purdue/xasm/c/sig/fold.c
/*	This is the convolution subroutine. It does summing
	knowing the length of the fcns and the sample rate.  */

fold(la,a,lb,b,lc,c,sample)
	float *a,*b,*c,sample;	int la,lb,lc;
{
	register i,j,k;
	lc = la + lb - 1;
	presto(lc,c,0.);
	for(i = 0;i != la;i++)
	    for(j = 0;j != lb;j++){
		k = i + j;
		c[k] =+ (a[i] * b[j])/sample;

		}
}

File: usenix80/toronto/purdue/xasm/c/sig/fold.f
	subroutine fold(la,a,lb,b,lc,c,sample)
	dimension a(1),b(1),c(1)
	real sample
	lc=la+lb-1
	call presto(lc,c,0.)
	do 10 i=1,la
	do 10 j=1,lb
	k=i+j-1
10	c(k)=c(k)+(a(i)*b(j))/sample
	return
	end
